Automattic/sensei
# Add extensions backend.

Fixes #5743

### Changes proposed in this Pull Request
* Create `Sensei_Home_Extensions_Provider`.
* Add `sensei_home_extensions` hook to extend extensions result.
* Cleanup a little bit Controller tests.

### Testing instructions
- Run: `curl -s -X GET --user aaron:development 'http://test2.local/wp-json/sensei-internal/v1/home' | jq .extensions`
- Confirm output makes sense.
- To test the hook you can use a snippet like: `add_filter( 'sensei_home_extensions', function() { return ['fake']; } );`.

### New/Updated Hooks
* Added `sensei_home_extensions` hook.

### Screenshot / Video
```
$ curl -s -X GET --user aaron:development 'http://test2.local/wp-json/sensei-internal/v1/home' | jq .extensions
[
  {
    "title": "Sensei LMS Post to Course Creator",
    "product_slug": "sensei-post-to-course",
    "description": "Turn your blog posts into online courses.",
    "image": "https://i0.wp.com/senseilms.com/wp-content/uploads/2022/02/sensei-post-to-course.png?resize=80%2C80&ssl=1",
    "price": 0,
    "more_url": "https://senseilms.com/product/sensei-lms-post-to-course-creator/"
  },
  {
    "title": "Sensei LMS Certificates",
    "product_slug": "sensei-certificates",
    "description": "Award your students with a certificate of completion and a sense of accomplishment after finishing a course.",
    "image": "https://i0.wp.com/senseilms.com/wp-content/uploads/2022/02/sensei-certificates.png?resize=80%2C80&ssl=1",
    "price": 0,
    "more_url": "https://senseilms.com/product/sensei-lms-certificates/"
  },
  {
    "title": "Sensei LMS Media Attachments",
    "product_slug": "sensei-media-attachments",
    "description": "Provide your students with easy access to additional learning materials, from audio files to slideshows and PDFs.",
    "image": "https://i0.wp.com/senseilms.com/wp-content/uploads/2022/02/sensei-media-attachments.png?resize=80%2C80&ssl=1",
    "price": 0,
    "more_url": "https://senseilms.com/product/sensei-lms-media-attachments/"
  }
]
```

## Notice ⚠️ 
- We might need to customise this a little bit further once we start implementing the frontend for this. I am quite open to change this interface and implementation to fit better whatever is needed in frontend.
